src/tests/unit/tests/DetailsView/components/save-assessment-button.test.tsx

```diff
@@ -36,71 +44,95 @@ describe('SaveAssessmentButton', () => {
             href: 'url',
             userConfigurationStoreData,
         };
-        eventStub = new EventStubFactory().createMouseClickEvent();
     });
 
     describe('on dialog enabled', () => {
-        let wrapper: ShallowWrapper;
+        describe('render', () => {
+            beforeEach(() => {
+                wrapper = render(<SaveAssessmentButton {...propsStub} />);
+                fireEvent.click(wrapper.getByRole('link'));
+            });
+            it('snapshot of dialog', () => {
+                wrapper = render(<SaveAssessmentButton {...propsStub} />);
 
-        beforeEach(() => {
-            wrapper = shallow(<SaveAssessmentButton {...propsStub} />);
-            wrapper.find(InsightsCommandButton).simulate('click', eventStub);
-        });
+                const dialog = wrapper.container.querySelector('mock-Dialog');
+                expect(dialog).toMatchSnapshot();
+            });
 
-        it('snapshot of dialog', () => {
-            expect(wrapper.getElement()).toMatchSnapshot();
-        });
+            it('dialog is visible', () => {
+                wrapper = render(<SaveAssessmentButton {...propsStub} />);
 
-        it('dialog is visible', () => {
-            expect(wrapper.find(Dialog).props().hidden).toEqual(false);
-        });
+                expect(getMockComponentClassPropsForCall(Dialog, 2).hidden).toEqual(false);
+            });
 
-        it('dialog is hidden (dismissed) when "got it" button is clicked', () => {
-            wrapper.find(PrimaryButton).simulate('click');
-            expect(wrapper.find(Dialog).props().hidden).toEqual(true);
-        });
+            it('dialog is hidden (dismissed) when onDismiss is called', () => {
+                wrapper = render(<SaveAssessmentButton {...propsStub} />);
```

Comment:
This is already rendered in the beforeEach. The 2 in the expect call refers to the rerender that happens when the link is clicked (also in the beforeEach.
```suggestion
```